src/pages/SignIn/SignIn.jsx

```diff
@@ -1,18 +1,49 @@
 import "./SignIn.css";
 import { useState, useCallback } from "react";
 import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import axios from "axios";
+import { usersActions } from "../../store/users-slice";
 import Input from "../../components/Input/Input";
 
 const SignIn = () => {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+
   const [isFormValid, setIsFormValid] = useState({
     username: false,
     password: false,
   });
+  const [userCredentials, setUserCredentials] = useState({
+    username: "",
+    password: "",
+  });
 
-  const goHomeHandler = (event) => {
+  //users: masha@gmail.com 1234masha
+  //       vasya@gmail.com 1234vasya
+  // admin: testAdmin@gmail.com 12345yuiopp
+  const signInHandler = (event) => {
     event.preventDefault();
-    navigate("/home");
+
+    axios
+      .post(
+        "https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyBAAugocQg-K1r5iMzzK1PLkqkn54flxTk",
+        {
+          email: userCredentials.username,
+          password: userCredentials.password,
+          returnSecureToken: true,
+        }
+      )
+      .then((response) => {
+        dispatch(
+          usersActions.login({
+            userEmail: response.data.email,
+            token: response.data.idToken,
+          })
+        );
+        navigate("/home");
+      })
+      .catch((error) => alert(error.message));
```

Comment:
Better handle this error this on page to show inline notification, not an alert window.